tests/mock_network/secure_message_delivery.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::{create_connected_nodes_until_split, poll_all};
+use itertools::Itertools;
+use routing::{
+    mock::Network, Authority, BlsPublicKeySet, FullId, HopMessage, Message, MessageContent,
+    RoutingMessage, SectionInfo, SignedRoutingMessage,
+};
+use std::collections::BTreeSet;
+
+#[test]
+#[should_panic]
+fn message_with_invalid_signature() {
+    let min_section_size = 3;
+    let network = Network::new(min_section_size, None);
+    let mut nodes = create_connected_nodes_until_split(&network, vec![1, 1], false);
+
+    let ori_info = unwrap!(nodes[0].inner.chain()).our_info().clone();
+    let ori_prefix = *ori_info.prefix();
+
+    let mut members = ori_info.members().clone();
+    let pub_id = *FullId::new().public_id();
+    let _ = members.insert(pub_id);
+
+    let new_info = unwrap!(SectionInfo::new_with_fields(
+        members.clone(),
+        ori_info.version() + 1,
+        ori_prefix,
+        BTreeSet::new(),
+    ));
+
+    let other_prefixes: Vec<_> = unwrap!(nodes[0].inner.chain())
+        .other_prefixes()
+        .iter()
+        .copied()
+        .collect_vec();
+    let src = Authority::Section(ori_prefix.name());
+    let dst = Authority::PrefixSection(other_prefixes[0]);
+    let content = MessageContent::NeighbourInfo(new_info.clone());
+
+    let routing_msg = RoutingMessage { src, dst, content };
+
+    let proof = unwrap!(nodes[0].inner.chain()).prove(&dst);
+    let pk_set = BlsPublicKeySet::from_section_info(new_info);
+    let signed_msg = unwrap!(SignedRoutingMessage::new(
```

Comment:
So this Create a Partial security_metadata.
Could probably just use a completely unrealted BLS Key that you create in the test instead of trying to get one from the node (should be simpler).

Reply:
just pushed a tryout commit which composes full security_metadata.
need to confirm whether the setup is correct